llvm/lib/Target/AMDGPU/VOP3Instructions.td

```diff
@@ -612,6 +622,10 @@ def : ThreeOp_i32_Pats<and, or, V_AND_OR_B32_e64>;
 def : ThreeOp_i32_Pats<or, or, V_OR3_B32_e64>;
 def : ThreeOp_i32_Pats<xor, add, V_XAD_U32_e64>;
 
+def : GCNPat<
+ (mul i32:$src0, IsPow2_32:$src1),
```

Comment:
As a general rule we should not select VALU instructions for uniform operations, so this should use `DivergentBinFrag<mul>`.